### PR TITLE
Wait for project to become available and use it when logging in as admin

### LINKED ISSUE #10274: flake in cmd/images: cannot delete istag
This one looks scary:

```
Running test/cmd/images_tests.sh:237: executing 'oc delete istag/perl:5.16 --context='default/127-0-0-1:28443/system:admin'' expecting success...
FAILURE after 0.213s: test/cmd/images_tests.sh:237: executing 'oc delete istag/perl:5.16 --context='default/127-0-0-1:28443/system:admin'' expecting success: the command returned the wrong error code
There was no output from the command.
Standard error from the command:
Error from server: imagestreams "perl" not found
```

Job https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_check/4708/console

Here's some test context:

```
# test deleting a tag using oc delete
os::cmd::expect_success_and_text "oc get is perl --template '{{(index .spec.tags 0).name}}'" '5.16'
os::cmd::expect_success_and_text "oc get is perl --template '{{(index .status.tags 0).tag}}'" 'latest'
os::cmd::expect_success "oc delete istag/perl:5.16 --context='${cluster_admin_context}'"
```

So it's clear that both image stream and the tag exist. Can the changed context be somehow related?

### PR DESCRIPTION
**Pull request by soltysh**
Fixes #10274

@mfojtik ptal

### CODE REVIEW

**Comment by soltysh** [comment]
[test]

**Comment by mfojtik** [comment]
@deads2k ptal

File: test/cmd/images_tests.sh [hunk 1]
```diff
@@ -238,7 +238,8 @@ os::test::junit::declare_suite_start "cmd/images${IMAGES_TESTS_POSTFIX:-}/delete
 # test deleting a tag using oc delete
 os::cmd::expect_success_and_text "oc get is perl --template '{{(index .spec.tags 0).name}}'" '5.16'
 os::cmd::expect_success_and_text "oc get is perl --template '{{(index .status.tags 0).tag}}'" 'latest'
-os::cmd::expect_success "oc delete istag/perl:5.16 --context='${cluster_admin_context}'"
+project="$( oc project -q )"
```
Comment (liggitt) [inline test/cmd/images_tests.sh]: Why was this using the cluster admin context in the first place? Does deleting an imagestream tag require cluster admin rights?

Comment (soltysh) [inline test/cmd/images_tests.sh]: Normally it does not require, but the [user created for this test](https://github.com/openshift/origin/blob/c97d6e7562b53ab0cf09f51219614d30d52a0592/test/cmd/images-old-policy.sh#L27) has no such powers.

**Comment by deads2k** [comment]
Not that I'm against flake fixes, but shouldn't we be more worried about why our context appears to be wrong?

File: test/cmd/images_tests.sh (hunk 1 above)
Comment (liggitt) [inline test/cmd/images_tests.sh]: if the cluster_admin_context was pointing to the default namespace, that means the project created for the test hadn't showed up in `list projects` yet when the cluster admin logged in at the top of the file

fix that by pinning the namespace in the `cluster_admin_context="$( oc config current-context )"` step

**Comment by soltysh** [comment]
@deads2k I think @liggitt explained it nicely. I'll apply Jordan's suggestion right away.

**Comment by mfojtik** [comment]
I agree with @deads2k. Fixing this is OK, but we need an issue to track why the context appears to be wrong.

**Comment by soltysh** [comment]
@liggitt changed to wait, lgty?

File: test/cmd/images_tests.sh [hunk 2]
```diff
@@ -21,7 +21,8 @@ os::test::junit::declare_suite_start "cmd/images${IMAGES_TESTS_POSTFIX:-}"
 # some steps below require that we use system:admin privileges, but we don't
 # want to stomp on whatever context we were given when we started
 original_context="$( oc config current-context )"
-os::cmd::expect_success 'oc login -u system:admin'
+os::cmd::try_until_text "oc get project" "${project}"
```
Comment (liggitt) [inline test/cmd/images_tests.sh]: shouldn't be needed... the things launching this script are already waiting until we have access

File: test/cmd/images_tests.sh [hunk 3]
```diff
@@ -21,7 +21,8 @@ os::test::junit::declare_suite_start "cmd/images${IMAGES_TESTS_POSTFIX:-}"
...
+os::cmd::expect_success "oc login -u system:admin -n '${project}'"
```
Comment (liggitt) [inline test/cmd/images_tests.sh]: also shouldn't be needed... my PR would already do all it could to stay in the same project

**Comment by liggitt** [comment]
I don't think this is needed once login remains in the current project (https://github.com/openshift/origin/pull/10378)

**Pull request closed without merging**